src/rules/prefer-native-locators.ts

```diff
@@ -0,0 +1,139 @@
+
+        const rolePattern = /^\[role=['"](.+?)['"]\]$/
+        const roleMatch = query.match(rolePattern)
+        if (roleMatch) {
+          context.report({
+            fix(fixer) {
+              const newText = `getByRole("${roleMatch[1]}")`
+              return fixer.replaceTextRange(rangeToReplace, newText)
+            },
+            messageId: 'unexpectedRoleQuery',
+            node,
+          })
+        }
+
+        const placeholderPattern = /^\[placeholder=['"](.+?)['"]\]$/
+        const placeholderMatch = query.match(placeholderPattern)
+        if (placeholderMatch) {
+          context.report({
+            fix(fixer) {
+              const newText = `getByPlaceholder("${placeholderMatch[1]}")`
+              return fixer.replaceTextRange(rangeToReplace, newText)
+            },
+            messageId: 'unexpectedPlaceholderQuery',
+            node,
+          })
+        }
+
+        const altTextPattern = /^\[alt=['"](.+?)['"]\]$/
+        const altTextMatch = query.match(altTextPattern)
+        if (altTextMatch) {
+          context.report({
+            fix(fixer) {
+              const newText = `getByAltText("${altTextMatch[1]}")`
+              return fixer.replaceTextRange(rangeToReplace, newText)
+            },
+            messageId: 'unexpectedAltTextQuery',
+            node,
+          })
+        }
+
+        const titlePattern = /^\[title=['"](.+?)['"]\]$/
+        const titleMatch = query.match(titlePattern)
+        if (titleMatch) {
+          context.report({
+            fix(fixer) {
+              const newText = `getByTitle("${titleMatch[1]}")`
+              return fixer.replaceTextRange(rangeToReplace, newText)
+            },
+            messageId: 'unexpectedTitleQuery',
+            node,
+          })
+        }
+
+        const testIdPattern = new RegExp(
+          `^\\[${testIdAttribute}=['"](.+?)['"]\\]`,
+        )
+        const testIdMatch = query.match(testIdPattern)
+        if (testIdMatch) {
+          context.report({
+            fix(fixer) {
+              const newText = `getByTestId("${testIdMatch[1]}")`
+              return fixer.replaceTextRange(rangeToReplace, newText)
+            },
+            messageId: 'unexpectedTestIdQuery',
+            node,
+          })
+        }
+      },
+    }
+  },
+  meta: {
+    docs: {
+      category: 'Best Practices',
+      description: 'Prefer native locator functions',
+      recommended: false,
+      url: 'https://github.com/playwright-community/eslint-plugin-playwright/tree/main/docs/rules/prefer-native-locators.md',
+    },
+    fixable: 'code',
+    messages: {
+      unexpectedAltTextQuery: 'Use .getByAltText() instead',
+      unexpectedLabelQuery: 'Use .getByLabel() instead',
+      unexpectedPlaceholderQuery: 'Use .getByPlaceholder() instead',
+      unexpectedRoleQuery: 'Use .getByRole() instead',
+      unexpectedTestIdQuery: 'Use .getByTestId() instead',
+      unexpectedTitleQuery: 'Use .getByTitle() instead',
```

Comment:
Let's drop the period in these messages

Reply:
https://github.com/playwright-community/eslint-plugin-playwright/pull/308/commits/c3b834db2654ddc594545ce9a28085ee0d44f344